yt/frontends/amrvac/data_structures.py

```diff
@@ -247,81 +242,50 @@ def _set_code_unit_attributes(self):
-                unit_length = 1
-                unit_numberdensity = 1
-                unit_temperature = 1
-                unit_velocity = 0
+                unit_length = (1, 'cm')
+                unit_numberdensity = (1, 'cm**-3')
+                unit_temperature = (1, 'K')
+                unit_velocity = (0, 'cm*s**-1')
+
+        # Convert to YT quantities
+        unit_length = self.quan(unit_length[0], unit_length[1])
+        unit_numberdensity = self.quan(unit_numberdensity[0], unit_numberdensity[1])
+        unit_temperature = self.quan(unit_temperature[0], unit_temperature[1])
+        unit_velocity = self.quan(unit_velocity[0], unit_velocity[1])
 
-        unit_density = (1.0 + 4.0*He_abundance) * mp * unit_numberdensity
+        He_abundance = 0.1  # hardcoded in AMRVAC
+
+        unit_density = (1.0 + 4.0*He_abundance) * mass_hydrogen_cgs * unit_numberdensity
         if unit_velocity == 0:
-            unit_pressure = (2.0 + 3.0*He_abundance) * unit_numberdensity * kb * unit_temperature
-            unit_velocity = np.sqrt(unit_pressure / unit_density)
+            unit_pressure = ((2.0 + 3.0*He_abundance) *
+                             unit_numberdensity * boltzmann_constant_cgs  * unit_temperature).to('dyn*cm**-2')
+            unit_velocity = (np.sqrt(unit_pressure / unit_density)).to('cm*s**-1')
         else:
-            unit_pressure = unit_density * unit_velocity**2
-            unit_temperature = unit_pressure / ((2.0 + 3.0*He_abundance) * unit_numberdensity * kb)
-        unit_magneticfield = np.sqrt(mu0 * unit_pressure)
-
-        # set units
-        if self._usr_unit_system == 'mks':
-            unit_density = self.quan(unit_density, 'kg*m**-3')
-            unit_length = self.quan(unit_length, 'm')
-            unit_numberdensity = self.quan(unit_numberdensity, 'm**-3')
-            unit_velocity = self.quan(unit_velocity, 'm*s**-1')
-            unit_pressure = self.quan(unit_pressure, 'Pa')
-            unit_magneticfield = self.quan(unit_magneticfield, 'T')
-        else:
-            unit_density = self.quan(unit_density, 'g*cm**-3')
-            unit_length = self.quan(unit_length, 'cm')
-            unit_numberdensity = self.quan(unit_numberdensity, 'cm**-3')
-            unit_velocity = self.quan(unit_velocity, 'cm*s**-1')
-            unit_pressure = self.quan(unit_pressure, 'dyn*cm**-2')
-            unit_magneticfield = self.quan(unit_magneticfield, 'gauss')
-        unit_temperature = self.quan(unit_temperature, 'K')
+            unit_pressure = (unit_density * unit_velocity**2).to('dyn*cm**-2')
+            unit_temperature = (unit_pressure /
+                                ((2.0 + 3.0*He_abundance) * unit_numberdensity * boltzmann_constant_cgs)).to('K')
         unit_time = unit_length / unit_velocity
         unit_mass = unit_density * unit_length**3
+        unit_magneticfield = (np.sqrt(4*np.pi * unit_pressure)).to('gauss')
 
         setdefaultattr(self, "length_unit", unit_length)
```

Comment:
I find it's confusing that you're using a different naming convention than yt given the end usage of your variables.
It would make a lot more sense to rename everything from `unit_x` to `x_unit`, as it's the name your variable end up being called by as dataset attributes anyway.